src/load.ts

```diff
@@ -38,34 +70,70 @@ export function useDefault(modules: any | any[]): any[] | any {
 }
 
 const load: Load = (function (): Load {
+	const resolver = typeof require.toUrl === 'function' ? require.toUrl :
+		typeof (<any> require).resolve === 'function' ? (<any> require).resolve :
+		(resourceId: string) => resourceId;
+
+	function pluginLoad(moduleIds: string[], load: Load, loader: (modulesIds: string[]) => Promise<any>) {
+		const pluginResourceIds: string[] = [];
+		moduleIds = moduleIds.map((id: string, i: number) => {
+			const parts = id.split('!');
+			pluginResourceIds[i] = parts[1];
+			return parts[0];
+		});
+
+		return loader(moduleIds).then((modules: any[]) => {
+			pluginResourceIds.forEach((resourceId: string, i: number) => {
+				if (typeof resourceId === 'string') {
+					const module = modules[i];
+					const defaultExport = module['default'] || module;
+
+					if (isPlugin(defaultExport)) {
+						resourceId = typeof defaultExport.normalize === 'function' ?
+							defaultExport.normalize(resourceId, resolver) :
+							resolver(resourceId);
+
+						modules[i] = defaultExport.load(resourceId, load);
+					}
+				}
+			});
+
+			return Promise.all(modules);
+		});
+	}
+
 	if (typeof module === 'object' && typeof module.exports === 'object') {
-		return function (contextualRequire: any, ...moduleIds: string[]): Promise<any[]> {
+		return function load(contextualRequire: any, ...moduleIds: string[]): Promise<any[]> {
 			if (typeof contextualRequire === 'string') {
 				moduleIds.unshift(contextualRequire);
 				contextualRequire = require;
 			}
-			return new Promise(function (resolve, reject) {
+
+			return pluginLoad(moduleIds, load, (moduleIds: string[]) => {
 				try {
-					resolve(moduleIds.map(function (moduleId): any {
-						return contextualRequire(moduleId);
+					return Promise.resolve(moduleIds.map(function (moduleId): any {
+						return contextualRequire(moduleId.split('!')[0]);
 					}));
 				}
 				catch (error) {
-					reject(error);
+					return Promise.reject(error);
 				}
 			});
 		};
 	}
 	else if (typeof define === 'function' && define.amd) {
-		return function (contextualRequire: any, ...moduleIds: string[]): Promise<any[]> {
+		return function load(contextualRequire: any, ...moduleIds: string[]): Promise<any[]> {
 			if (typeof contextualRequire === 'string') {
 				moduleIds.unshift(contextualRequire);
 				contextualRequire = require;
 			}
-			return new Promise(function (resolve) {
-				// TODO: Error path once https://github.com/dojo/loader/issues/14 is figured out
-				contextualRequire(moduleIds, function (...modules: any[]) {
-					resolve(modules);
+
+			return pluginLoad(moduleIds, load, (moduleIds: string[]) => {
+				return new Promise(function (resolve, reject) {
+					// TODO: Error path once https://github.com/dojo/loader/issues/14 is figured out
```

Comment:
Yes, but doing so seems tangential to this particular PR. I'm more than happy to open another issue/PR specific to this, however.